### ISSUE TITLE
List of files uploaded 

### ISSUE THREAD

**Issue by wajaja** [1up-lab/OneupUploaderBundle#187]
Hey;

I have a problem in my EventListener;
uploaded files are in the following directory. \web\uploads\gallery
but I do not seem to get the list of uploaded files and their respective names in my EventListener

here is my eventlistener.php file

//...
class UploadListener
{
    public function __construct($doctrine)
    {
        $this->doctrine = $doctrine;
    }

    public function onUpload(PostPersistEvent $event)
    {
        $file = $event->getFile();
        //For image type
        
        $image = new Image();
        $image->setUrl($file->getPathName());

        
        $em = $this->getDoctrine()->getManager();
        $em->persist($image);
        $em->flush();
        
        //...

**Comment by ahmetalsan** [comment]
yes i need to get uploaded files too

**Comment by lsv** [comment]
TL;DR

````
/** @var \Symfony\Component\HttpFoundation\File\UploadedFile $originalFile */
$originalFile = $event->getRequest()->files->get('file');
````

And to show a use case

In my upload form, I have a checkbox for either rename the original filename or overwrite.
If the checkbox is only rename, and the file (client original name) exists, then the ```onValidate``` throws a exception.

The ```onUpload``` if the file (client original name) already exists then the original name will be changed.

The client original name is stored in my entity and a unique namer is used to actually save the file.
The client original name is used to include the file in the CMS

````php
<?php
namespace CmsBundle\MediaUploader;

use Oneup\UploaderBundle\Event\PostPersistEvent;
use Oneup\UploaderBundle\Event\ValidationEvent;
use Oneup\UploaderBundle\Uploader\Exception\ValidationException;
use CmsBundle\Entity\Media;
use CmsBundle\Repository\MediaRepository;
use Symfony\Component\EventDispatcher\EventSubscriberInterface;
use Symfony\Component\Translation\TranslatorInterface;

class MediaUploaderListener implements EventSubscriberInterface
{

    /**
     * @var MediaRepository
     */
    private $repository;

    /**
     * @var TranslatorInterface
     */
    private $translator;

    public function __construct(MediaRepository $repository, TranslatorInterface $translator)
    {
        $this->repository = $repository;
        $this->translator = $translator;
    }

    public function onValidate(ValidationEvent $event)
    {
        /** @var \Oneup\UploaderBundle\Uploader\File\FilesystemFile $file */
        $file = $event->getFile();
        $mediafile = $this->findFile($file->getClientOriginalName());

        if (! $event->getRequest()->request->has('overwrite')) {
            if ($mediafile) {
                throw new ValidationException($this->translator->trans('File already exists'));
            }
        }
    }

    public function onUpload(PostPersistEvent $event)
    {
        /** @var \Symfony\Component\HttpFoundation\File\UploadedFile $originalFile */
        $originalFile = $event->getRequest()->files->get('file');

        /** @var \Symfony\Component\HttpFoundation\File\File $file */
        $file = $event->getFile();

        $mediafile = $this->findFile($originalFile->getClientOriginalName());
        if ($mediafile) {
            // Rename this file
            $clientName = $this->getRenamedFilename(
                $originalFile->getClientOriginalName(),
                $originalFile->getClientOriginalExtension()
            );
        } else {
            $clientName = $originalFile->getClientOriginalName();
        }

        // Save it
        $obj = new Media();
        $obj->setExtension($originalFile->getClientOriginalExtension());
        $obj->setOriginalName($clientName);
        $obj->setPath($file->getBasename());
        $this->repository->getEm()->persist($obj);
        $this->repository->getEm()->flush();

    }

    /**
     * @param string $filename
     * @param string $extension
     * @param int $num
     * @return string
     */
    private function getRenamedFilename($filename, $extension, $num = 1)
    {
        $name = sprintf(
            '%s_%d.%s',
            substr_replace($filename, '', -(strlen($extension) + 1)),
            $num,
            $extension
        );

        if ($this->findFile($name)) {
            return $this->getRenamedFilename($filename, $extension, ++$num);
        }

        return $name;
    }

    /**
     * @param string $filename
     * @return null|Media
     */
    private function findFile($filename)
    {
        return $this->repository->findOneBy(['originalName' => $filename]);
    }

    /**
     * Returns an array of event names this subscriber wants to listen to.
     *
     * @return array The event names to listen to
     */
    public static function getSubscribedEvents()
    {
        return [
            'oneup_uploader.validation' => 'onValidate',
            'oneup_uploader.post_persist' => 'onUpload'
        ];
    }
}
````
